### PR TITLE
boolean support added

### PR DESCRIPTION
**Pull request by sivankumar86**
Now you can query boolean datatype from dynamodb

### CODE REVIEW

**Comment by sivankumar86** [comment]
https://patch-diff.githubusercontent.com/raw/awslabs/emr-dynamodb-connector/pull/38.patch

File: emr-dynamodb-hadoop/src/main/java/org/apache/hadoop/dynamodb/key/DynamoDBBooleanKey.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package org.apache.hadoop.dynamodb.key;
+
+import com.amazonaws.services.dynamodbv2.model.AttributeValue;
+
+/**
```
Comment (ajay25) [inline emr-dynamodb-hadoop/src/main/java/org/apache/hadoop/dynamodb/key/DynamoDBBooleanKey.java]: Remove the author comments for all the files

File: emr-dynamodb-hadoop/src/main/java/org/apache/hadoop/dynamodb/key/DynamoDBBooleanKey.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+ * Created by srramas on 7/26/17.
+ */
+public class DynamoDBBooleanKey  extends AbstractDynamoDBKey {
+
+    public DynamoDBBooleanKey(String key) {
+        super(key);
+    }
+    @Override
```
Comment (ajay25) [inline emr-dynamodb-hadoop/src/main/java/org/apache/hadoop/dynamodb/key/DynamoDBBooleanKey.java]: Missing line break

File: emr-dynamodb-hadoop/src/main/java/org/apache/hadoop/dynamodb/key/DynamoDBBooleanKey.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package org.apache.hadoop.dynamodb.key;
```
Comment (ajay25) [inline emr-dynamodb-hadoop/src/main/java/org/apache/hadoop/dynamodb/key/DynamoDBBooleanKey.java]: Missing the copyright header even though I'm not sure whether it is still needed. Someone else can confirm this.

File: emr-dynamodb-hive/src/main/java/org/apache/hadoop/hive/dynamodb/type/HiveDynamoDBTypeFactory.java [hunk 4]
```diff
@@ -42,6 +44,8 @@
     HIVE_TYPE_MAP.put(serdeConstants.DOUBLE_TYPE_NAME, NUMBER_TYPE);
     HIVE_TYPE_MAP.put(serdeConstants.BIGINT_TYPE_NAME, NUMBER_TYPE);
     HIVE_TYPE_MAP.put(serdeConstants.BINARY_TYPE_NAME, BINARY_TYPE);
+    HIVE_TYPE_MAP.put(serdeConstants.BOOLEAN_TYPE_NAME, BOOLEAN_TYPE); //boolean support added
```
Comment (ajay25) [inline emr-dynamodb-hive/src/main/java/org/apache/hadoop/hive/dynamodb/type/HiveDynamoDBTypeFactory.java]: get rid of the comment

File: emr-dynamodb-hive/src/main/java/org/apache/hadoop/hive/dynamodb/type/HiveDynamoDBBooleanType.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
+package org.apache.hadoop.hive.dynamodb.type;
+
+import com.amazonaws.services.dynamodbv2.model.AttributeValue;
+import org.apache.hadoop.dynamodb.key.DynamoDBKey;
```
Comment (ajay25) [inline emr-dynamodb-hive/src/main/java/org/apache/hadoop/hive/dynamodb/type/HiveDynamoDBBooleanType.java]: line break

File: emr-dynamodb-hadoop/src/main/java/org/apache/hadoop/dynamodb/type/DynamoDBBooleanType.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
+package org.apache.hadoop.dynamodb.type;
+
+import org.apache.hadoop.dynamodb.key.DynamoDBBooleanKey;
+import org.apache.hadoop.dynamodb.key.DynamoDBKey;
+import com.amazonaws.services.dynamodbv2.model.AttributeValue;
```
Comment (ajay25) [inline emr-dynamodb-hadoop/src/main/java/org/apache/hadoop/dynamodb/type/DynamoDBBooleanType.java]: move this above other imports

File: emr-dynamodb-hive/src/main/java/org/apache/hadoop/hive/dynamodb/util/DynamoDBDataParser.java [hunk 7]
```diff
@@ -40,6 +37,15 @@ public String getNumber(Object data, ObjectInspector objectInspector) {
     }
   }
 
+    public Boolean getBoolean(Object data, ObjectInspector objectInspector) {
+        if (objectInspector instanceof BooleanObjectInspector) {
+            return ((BooleanObjectInspector)objectInspector).get(data);
+        }else {
```
Comment (ajay25) [inline emr-dynamodb-hive/src/main/java/org/apache/hadoop/hive/dynamodb/util/DynamoDBDataParser.java]: else seems redundant here

File: emr-dynamodb-hive/src/main/java/org/apache/hadoop/hive/dynamodb/util/DynamoDBDataParser.java [hunk 8]
```diff
@@ -40,6 +37,15 @@ public String getNumber(Object data, ObjectInspector objectInspector) {
     }
   }
 
+    public Boolean getBoolean(Object data, ObjectInspector objectInspector) {
+        if (objectInspector instanceof BooleanObjectInspector) {
+            return ((BooleanObjectInspector)objectInspector).get(data);
```
Comment (ajay25) [inline emr-dynamodb-hive/src/main/java/org/apache/hadoop/hive/dynamodb/util/DynamoDBDataParser.java]: minor: space between (BooleanObjectInspector) and objectInspector

File: emr-dynamodb-hadoop/src/main/java/org/apache/hadoop/dynamodb/key/DynamoDBBooleanKey.java (hunk 3 above)
Comment (sivankumar86) [inline emr-dynamodb-hadoop/src/main/java/org/apache/hadoop/dynamodb/key/DynamoDBBooleanKey.java]: Copied copyright header from other class

File: emr-dynamodb-hadoop/src/main/java/org/apache/hadoop/dynamodb/key/DynamoDBBooleanKey.java (hunk 2 above)
Comment (sivankumar86) [inline emr-dynamodb-hadoop/src/main/java/org/apache/hadoop/dynamodb/key/DynamoDBBooleanKey.java]: added line break

File: emr-dynamodb-hadoop/src/main/java/org/apache/hadoop/dynamodb/key/DynamoDBBooleanKey.java (hunk 1 above)
Comment (sivankumar86) [inline emr-dynamodb-hadoop/src/main/java/org/apache/hadoop/dynamodb/key/DynamoDBBooleanKey.java]: done

File: emr-dynamodb-hive/src/main/java/org/apache/hadoop/hive/dynamodb/util/DynamoDBDataParser.java [hunk 9]
```diff
@@ -17,10 +17,7 @@
 import org.apache.hadoop.hive.serde.Constants;
 import org.apache.hadoop.hive.serde2.objectinspector.ListObjectInspector;
 import org.apache.hadoop.hive.serde2.objectinspector.ObjectInspector;
-import org.apache.hadoop.hive.serde2.objectinspector.primitive.BinaryObjectInspector;
-import org.apache.hadoop.hive.serde2.objectinspector.primitive.DoubleObjectInspector;
-import org.apache.hadoop.hive.serde2.objectinspector.primitive.LongObjectInspector;
-import org.apache.hadoop.hive.serde2.objectinspector.primitive.StringObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.primitive.*;
```
Comment (itsgrimetime) [inline emr-dynamodb-hive/src/main/java/org/apache/hadoop/hive/dynamodb/util/DynamoDBDataParser.java]: Please keep individual explicit imports

Comment (sivankumar86) [reply emr-dynamodb-hive/src/main/java/org/apache/hadoop/hive/dynamodb/util/DynamoDBDataParser.java]: Thanks for reviewing. modified the code as per review comments.

**Comment by sivankumar86** [comment]
changes has been made

**Comment by dtabolich** [comment]
Hi, where you're with this merge?
It would be really great to get out of the box support for BOOL.

Thanks.

**Comment by itsgrimetime** [comment]
It looks good to me - @ajay25 do you have any more concerns?

**Comment by ajay25** [comment]
lgtm 👍

**Comment by dtabolich** [comment]
Folks,

I'd appreciate if you can help me to understand how to reference new version from EMR? I mean, based on my understanding, this emr connector is shipped as a part of EMR release. So, I am just curios if there is an option to build master and register this new *.jar instead of currently available one. We use Data Pipeline HiveActivity to export Dynamo data.

Thanks.

**Pull request merged**